vmdb/app/models/metric/ci_mixin/state_finders.rb

```diff
@@ -26,7 +26,7 @@ def vim_performance_state_for_ts(ts)
   end
 
   def preload_vim_performance_state_for_ts(conditions = {})
-    @states_by_ts = self.vim_performance_states.all(:conditions => conditions).index_by(&:timestamp)
+    @states_by_ts = self.vim_performance_states.where(conditions).index_by(&:timestamp)
```

Comment:
If conditions is an empty Hash as per the default for the param, will that cause a problem?